--- a/HaXml.cabal
+++ b/HaXml.cabal
@@ -66,68 +66,68 @@ library
   default-extensions: CPP, ExistentialQuantification
   nhc98-options: -K10M
   if impl(ghc >= 8.2)
-    ghc-options:       -fhide-source-paths
+    ghc-options:       -fhide-source-paths -no-pie
 
 Executable Canonicalise
-  GHC-Options: -Wall
+  GHC-Options: -Wall -optc-no-pie -optl-no-pie -optP-no-pie
   Default-Language: Haskell98
   Default-Extensions:  CPP
   Hs-Source-Dirs: src/tools
   Main-Is: Canonicalise.hs
   build-depends: base, HaXml, pretty
 
 Executable CanonicaliseLazy
-  GHC-Options: -Wall
+  GHC-Options: -Wall -optc-no-pie -optl-no-pie -optP-no-pie
   Default-Language: Haskell98
   Default-Extensions:  CPP
   Hs-Source-Dirs: src/tools
   Main-Is: CanonicaliseLazy.hs
   build-depends: base, HaXml, pretty
 
 Executable Xtract
-  GHC-Options: -Wall
+  GHC-Options: -Wall -optc-no-pie -optl-no-pie -optP-no-pie
   Default-Language: Haskell98
   Default-Extensions:  CPP
   Hs-Source-Dirs: src/tools
   Main-Is: Xtract.hs
   build-depends: base, HaXml, pretty
 
 Executable Validate
-  GHC-Options: -Wall
+  GHC-Options: -Wall -optc-no-pie -optl-no-pie -optP-no-pie
   Default-Language: Haskell98
   Default-Extensions:  CPP
   Hs-Source-Dirs: src/tools
   Main-Is: Validate.hs
   build-depends: base, HaXml
 
 Executable MkOneOf
-  GHC-Options: -Wall
+  GHC-Options: -Wall -optc-no-pie -optl-no-pie -optP-no-pie
   Default-Language: Haskell98
   Default-Extensions:  CPP
   Hs-Source-Dirs: src/tools
   Main-Is: MkOneOf.hs
   build-depends: base, HaXml
 
 Executable DtdToHaskell
-  GHC-Options: -Wall
+  GHC-Options: -Wall -optc-no-pie -optl-no-pie -optP-no-pie
   Default-Language: Haskell98
   Default-Extensions:  CPP
   Hs-Source-Dirs: src/tools
   Main-Is: DtdToHaskell.hs
   build-depends: base, HaXml, pretty
 
 Executable XsdToHaskell
-  GHC-Options: -Wall
+  GHC-Options: -Wall -optc-no-pie -optl-no-pie -optP-no-pie
   Default-Language: Haskell98
   Default-Extensions:  CPP
   Hs-Source-Dirs: src/tools
   Main-Is: XsdToHaskell.hs
   build-depends: base, HaXml, pretty, polyparse, directory
 
 Executable FpMLToHaskell
-  GHC-Options: -Wall
+  GHC-Options: -Wall -optc-no-pie -optl-no-pie -optP-no-pie
   Default-Language: Haskell98
   Default-Extensions:  CPP
   Hs-Source-Dirs: src/tools
   Main-Is: FpMLToHaskell.hs
   build-depends: base, HaXml, pretty, polyparse, directory
