--- a/HaXml.cabal
+++ b/HaXml.cabal
@@ -66,50 +66,50 @@ library
   nhc98-options: -K10M
 
 Executable Canonicalise
-  GHC-Options: -Wall
+  GHC-Options: -Wall -optc-no-pie -optl-no-pie -optP-no-pie
   Extensions:  CPP
   Hs-Source-Dirs: src/tools, src
-  cpp-options: -DMYVERSION="1.20.2"
+  cpp-options: -DMYVERSION="1.20.2" -no-pie
   Main-Is: Canonicalise.hs
 
 Executable CanonicaliseLazy
-  GHC-Options: -Wall
+  GHC-Options: -Wall -optc-no-pie -optl-no-pie -optP-no-pie
   Extensions:  CPP
   Hs-Source-Dirs: src/tools, src
-  cpp-options: -DMYVERSION="1.20.2"
+  cpp-options: -DMYVERSION="1.20.2" -no-pie
   Main-Is: CanonicaliseLazy.hs
 
 Executable Xtract
-  GHC-Options: -Wall
+  GHC-Options: -Wall -optc-no-pie -optl-no-pie -optP-no-pie
   Extensions:  CPP
   Hs-Source-Dirs: src/tools, src
-  cpp-options: -DMYVERSION="1.20.2"
+  cpp-options: -DMYVERSION="1.20.2" -no-pie
   Main-Is: Xtract.hs
 
 Executable XtractLazy
-  GHC-Options: -Wall
+  GHC-Options: -Wall -optc-no-pie -optl-no-pie -optP-no-pie
   Extensions:  CPP
   Hs-Source-Dirs: src/tools, src
-  cpp-options: -DMYVERSION="1.20.2"
+  cpp-options: -DMYVERSION="1.20.2" -no-pie
   Main-Is: XtractLazy.hs
 
 Executable Validate
-  GHC-Options: -Wall
+  GHC-Options: -Wall -optc-no-pie -optl-no-pie -optP-no-pie
   Extensions:  CPP
   Hs-Source-Dirs: src/tools, src
-  cpp-options: -DMYVERSION="1.20.2"
+  cpp-options: -DMYVERSION="1.20.2" -no-pie
   Main-Is: Validate.hs
 
 Executable MkOneOf
-  GHC-Options: -Wall
+  GHC-Options: -Wall -optc-no-pie -optl-no-pie -optP-no-pie
   Extensions:  CPP
   Hs-Source-Dirs: src/tools, src
-  cpp-options: -DMYVERSION="1.20.2"
+  cpp-options: -DMYVERSION="1.20.2" -no-pie
   Main-Is: MkOneOf.hs
 
 Executable DtdToHaskell
-  GHC-Options: -Wall
+  GHC-Options: -Wall -optc-no-pie -optl-no-pie -optP-no-pie
   Extensions:  CPP
   Hs-Source-Dirs: src/tools, src
-  cpp-options: -DMYVERSION="1.20.2"
+  cpp-options: -DMYVERSION="1.20.2" -no-pie
   Main-Is: DtdToHaskell.hs
