commit 0e0130b74f36a2571fed5123c853163ca01ad7ae
Author: Malcolm.Wallace@cs.york.ac.uk <unknown>
Date:   Tue Dec 1 03:37:45 2009 +0000

    x86_64 requires more stack

diff --git a/libraries/base/Makefile.nhc98 b/libraries/base/Makefile.nhc98
index a3e6fc43f2..d11e1518df 100644
--- a/libraries/base/Makefile.nhc98
+++ b/libraries/base/Makefile.nhc98
@@ -1,7 +1,7 @@
 THISPKG	= base
 SEARCH	= -I$(TOPDIR)/targets/$(MACHINE) -Iinclude \
 	  -I../../prelude/PreludeIO -I../../prelude/`$(LOCAL)harch`
-EXTRA_H_FLAGS   = -H4M -K3M
+EXTRA_H_FLAGS   = -H4M -K6M
 EXTRA_C_FLAGS   = -D__NHC__
 EXTRA_HBC_FLAGS = -H16M -A1M
 
