Index: ghc-8.8.4/rts/Printer.c
===================================================================
--- ghc-8.8.4.orig/rts/Printer.c
+++ ghc-8.8.4/rts/Printer.c
@@ -808,6 +808,12 @@ static void printZcoded( const char *raw
 */
 #ifdef USING_LIBBFD
 
+#ifndef PACKAGE
+# define PACKAGE "ghc"
+#endif
+#ifndef PACKAGE_VERSION
+# define PACKAGE_VERSION "8.8.4"
+#endif
 #include <bfd.h>
 
 /* Fairly ad-hoc piece of code that seems to filter out a lot of
