--- a/conduit.cabal
+++ b/conduit.cabal
@@ -44,7 +44,7 @@ Library
                        Data.Streaming.FileRead
                        Data.Streaming.Filesystem
   Build-depends:       base                     >= 4.12         && < 5
-                     , resourcet                >= 1.2          && < 1.3
+                     , resourcet                >= 1.2          && < 1.4
                      , transformers             >= 0.4
                      , mtl
                      , primitive
